persistence/sql/persister_consent.go

```diff
@@ -158,51 +160,46 @@ func (p *Persister) GetForcedObfuscatedLoginSession(ctx context.Context, client,
 // CreateConsentRequest configures fields that are introduced or changed in the
 // consent request. It doesn't touch fields that would be copied from the login
 // request.
-func (p *Persister) CreateConsentRequest(ctx context.Context, req *consent.OAuth2ConsentRequest) error {
+func (p *Persister) CreateConsentRequest(ctx context.Context, f *flow.Flow, req *flow.OAuth2ConsentRequest) error {
 	ctx, span := p.r.Tracer(ctx).Tracer().Start(ctx, "persistence.sql.CreateConsentRequest")
 	defer span.End()
 
-	c, err := p.Connection(ctx).RawQuery(`
-UPDATE hydra_oauth2_flow
-SET
-	state = ?,
-	consent_challenge_id = ?,
-	consent_skip = ?,
-	consent_verifier = ?,
-	consent_csrf = ?
-WHERE login_challenge = ? AND nid = ?;
-`,
-		flow.FlowStateConsentInitialized,
-		sqlxx.NullString(req.ID),
-		req.Skip,
-		req.Verifier,
-		req.CSRF,
-		req.LoginChallenge.String(),
-		p.NetworkID(ctx),
-	).ExecWithCount()
-	if err != nil {
-		return sqlcon.HandleError(err)
+	if f == nil {
+		return errorsx.WithStack(x.ErrNotFound.WithDebug("Flow is nil"))
 	}
-	if c != 1 {
+	if f.ID != req.LoginChallenge.String() || f.NID != p.NetworkID(ctx) {
 		return errorsx.WithStack(x.ErrNotFound)
 	}
+	f.State = flow.FlowStateConsentInitialized
+	f.ConsentChallengeID = sqlxx.NullString(req.ID)
+	f.ConsentSkip = req.Skip
+	f.ConsentVerifier = sqlxx.NullString(req.Verifier)
+	f.ConsentCSRF = sqlxx.NullString(req.CSRF)
+
 	return nil
 }
 
 func (p *Persister) GetFlowByConsentChallenge(ctx context.Context, challenge string) (*flow.Flow, error) {
 	ctx, span := p.r.Tracer(ctx).Tracer().Start(ctx, "persistence.sql.GetFlowByConsentChallenge")
 	defer span.End()
 
-	f := &flow.Flow{}
-
-	if err := sqlcon.HandleError(p.QueryWithNetwork(ctx).Where("consent_challenge_id = ?", challenge).First(f)); err != nil {
-		return nil, err
+	// challenge contains the flow.
+	f, err := flowctx.Decode[flow.Flow](ctx, p.r.FlowCipher(), challenge, flowctx.AsConsentChallenge)
+	if err != nil {
+		return nil, errorsx.WithStack(x.ErrNotFound)
+	}
+	if f.NID != p.NetworkID(ctx) {
+		return nil, errorsx.WithStack(x.ErrNotFound)
+	}
+	// TODO(hperl): Always check for expiration when decoding the flow.
+	if f.RequestedAt.Add(p.config.ConsentRequestMaxAge(ctx)).Before(time.Now()) {
```

Comment:
Why do we need to check the expiry here, when we did not check it before in the method? Is there a particular reason for it?

Reply:
I don't understand the difference here.

Reply:
IMHO because it makes it easier to reason about expiry properties. We never handle expired data.